WordPress/Classes/Utility/Environment/Environment.swift

```diff
@@ -0,0 +1,47 @@
+
+import Foundation
+
+/// A collection of global variables and singletons that the app wants access to.
+///
+struct Environment {
+
+    // MARK: - Globals
+
+    /// A type that helps tracking whether or not a user should be prompted for an app review
+    let appRatingUtility: AppRatingUtilityType
+
+    /// The main Core Data context
+    let mainContext: NSManagedObjectContext
```

Comment:
I wonder if the Environment should provide the `ContextManager`, instead of the context itself 🤔 

The call site would then become: `Environment.current.contextManager.mainContext`.

Reply:
I was thinking the same thing. `ContextManager` has methods that we will need to be able to test anyway. The `Environment.current.contextManager.mainContext` (so long) was what made me rethink that.

Do you think it is just fine? 

As an alternative we could add a helper computed var to access `mainContext` directly, so we can keep `Environment.current.mainContext`.

Reply:
Caveat here is: the Services architecture requires you to... well, pass over a MOC during initialization. If you wanted to go BG, you'd find yourself recreating some of the goodies we've got in ContextManager. 

Alternatively: You can always do...

```
let contextManager: ContextManager

var mainContext: NSManagedObjectContext {
   return contextManager.mainContext
}
```